ibm/resource_ibm_cloudant.go

```diff
@@ -0,0 +1,603 @@
+	capacity := d.Get("capacity").(int)
+	if capacity > 1 && plan == "lite" {
+		return fmt.Errorf("Setting capacity is not supported for your instance's plan.")
+	}
+	return nil
+}
+
+func setCloudantInstanceCapacity(client *cloudantv1.CloudantV1, d *schema.ResourceData) error {
+	capacityThroughputInformation, err := readCloudantInstanceCapacity(client)
+	if err != nil {
+		return fmt.Errorf("Error retrieving capacity throughput information: %s", err)
+	}
+
+	if capacityThroughputInformation.Current != nil && capacityThroughputInformation.Current.Throughput != nil {
+		currentThroughput := capacityThroughputInformation.Current.Throughput
+		// lite plan doesn't have "blocks" attr on broker's response
+		if d.Get("plan").(string) == "lite" {
+			d.Set("capacity", 1)
+		} else {
+			blocks := int(*currentThroughput.Blocks)
+			d.Set("capacity", blocks)
+		}
+		throughput := map[string]int{
+			"query": int(*currentThroughput.Query),
+			"read":  int(*currentThroughput.Read),
+			"write": int(*currentThroughput.Write),
+		}
+		d.Set("throughput", throughput)
+	}
+	return nil
+}
+
+func readCloudantInstanceCapacity(client *cloudantv1.CloudantV1) (*cloudantv1.CapacityThroughputInformation, error) {
+	opts := client.NewGetCapacityThroughputInformationOptions()
+
+	capacityThroughputInformation, response, err := client.GetCapacityThroughputInformation(opts)
+	if err != nil {
+		log.Printf("[DEBUG] Error getting capacity throughput information: %s\n%s", err, response)
+	}
+	return capacityThroughputInformation, nil
+}
+
+func updateCloudantInstanceCapacity(client *cloudantv1.CloudantV1, d *schema.ResourceData) error {
+	blocks := int64(d.Get("capacity").(int))
+
+	putOpts := client.NewPutCapacityThroughputConfigurationOptions(blocks)
+
+	_, response, err := client.PutCapacityThroughputConfiguration(putOpts)
+	if err != nil {
+		log.Printf("[DEBUG] Error updating capacity throughput: %s\n%s", err, response)
+		return err
+	}
+
+	return isWaitForCapacityUpdated(client)
+}
+
+func isWaitForCapacityUpdated(client *cloudantv1.CloudantV1) error {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"retry"},
+		Target:  []string{"done", "failed"},
+		Refresh: func() (interface{}, string, error) {
+			capacityThroughputInformation, err := readCloudantInstanceCapacity(client)
+			if err != nil {
+				return nil, "failed", err
+			}
+
+			state := "retry"
+			current := *capacityThroughputInformation.Current.Throughput.Blocks
+			target := *capacityThroughputInformation.Target.Throughput.Blocks
+
+			if current == target {
+				state = "done"
+			}
+
+			return current, state, nil
+		},
+		Timeout:    5 * time.Minute,
+		Delay:      5 * time.Second,
+		MinTimeout: 2 * time.Second,
+	}
+
+	_, err := stateConf.WaitForState()
+	return err
+}
+
+func validateCloudantInstanceCors(d *schema.ResourceData) error {
+	enableCors := d.Get("enable_cors").(bool)
+	corsConfigRaw := d.Get("cors_config").([]interface{})
+	if !enableCors && len(corsConfigRaw) > 0 {
```

Comment:
Why is the `enableCors` in negation? Should the `true` mean turning on CORS?
```suggestion
	if enableCors && len(corsConfigRaw) > 0 {
```